### PR TITLE
Add game preset attribute, unhardcode game presets

### PR DESCRIPTION
**Pull request by DrSmugleaf**
## About the PR
The death of the game preset switch expression has come.
Now game presets are marked with `[GamePreset(id, aliases...)]`

### CODE REVIEW

File: Content.Server/Players/PlayerData.cs [hunk 1]
```diff
@@ -60,5 +61,15 @@ public static class PlayerDataExt
...
+
+        public static bool TryContentData(this IPlayerData engineData, [NotNullWhen(true)] out PlayerData? data)
+        {
+            return (data = engineData.ContentData()) != null;
+        }
+
+        public static bool TryContentData(this IPlayerSession session, [NotNullWhen(true)] out PlayerData? data)
+        {
+            return (data = session.ContentData()) != null;
+        }
```
Comment (metalgearsloth) [inline Content.Server/Players/PlayerData.cs:73]: I see you sneakin

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
nullables

File: Content.Server/Players/PlayerData.cs (hunk 1 above)
Comment (PaulRitter) [reply Content.Server/Players/PlayerData.cs:73]: thistbh, fix dis & get merged

**Review by PaulRitter — APPROVED** [review]
based based based

**Review by PaulRitter — CHANGES_REQUESTED** [review]
wrong button

File: Content.Server/Players/PlayerData.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Server/Players/PlayerData.cs]: Done.

**Review by PaulRitter — APPROVED** [review]

**Pull request merged**